.github/workflows/commit_command_checker.yaml

```diff
@@ -0,0 +1,35 @@
+name: 'Check that commit commands have been used correctly'
+on:
+  workflow_call:
+
+jobs:
+  commit-command-checker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check that 'main' is not merged into the development branch
+        uses: gsactions/commit-message-checker@v2
+        with:
+          excludeDescription: 'true'  # optional: this excludes the description body of a pull request
+          excludeTitle: 'true'  # optional: this excludes the title of a pull request
+          checkAllCommitMessages: 'true'  # optional: this checks all commits associated with a pull request
+          accessToken: ${{ secrets.GITHUB_TOKEN }}  # github access token is only required if checkAllCommitMessages is true
+          pattern: ^(?!Merge branch 'main')
+          error: |
+            "This step failed because you merged the 'main' branch into the development branch,
+             likely by clicking the 'Update my branch' button on the GitHub pull request page.
+             To avoid this error, update the 'main' branch locally and rebase against it.
+             See https://developer.lsst.io/work/flow.html#pushing-code for detailed instructions."
+      - name: Check the divergent remote and local branches were merged
+        uses: gsactions/commit-message-checker@v2
+        if: ${{ success() }}
```

Comment:
The more generic second step runs only if the first step passes. This is to avoid multiple error messages for a single error (merging 'main' to the ticket branch).